src/Education.UnitTests/SonarRuleIdUriEncoderDecoderTests.cs

```diff
@@ -36,5 +37,30 @@ public void EncodeToUri_EncodesCompositeKeyCorrectly()
 
             uri.AbsoluteUri.Should().Be("sonarlintrulecrossref://cpp/S1234");
         }
+
+        [TestMethod]
+        public void TryDecodeToCompositeRuleId_IncomingUriCanBeDecoded_ReturnsTrue()
+        {
+            var decodableUri = new Uri($"sonarlintrulecrossref://cpp/S1234");
+
+            var result = SonarRuleIdUriEncoderDecoder.TryDecodeToCompositeRuleId(decodableUri, out SonarCompositeRuleId compositeRuleId);
+
+            result.Should().BeTrue();
+            compositeRuleId.RepoKey.Should().Be("cpp");
+            compositeRuleId.RuleKey.Should().Be("S1234");
+        }
+
+        [TestMethod]
```

Comment:
Two test cases so its clear that it doesn't need to be in https format to fail, it just needs to be anything BUT the correct prefix